### ISSUE TITLE
Optimize out empty If statements in LIR

### ISSUE THREAD

**Issue by justinkambic** [elastic/logstash#9301]
Presently Logstash allows the creation of pipelines with empty `if` statements, i.e. a condition is provided and no plugins or subsequent `if` statements are specified within its scope.

```
input {
  stdin { }
}

filter {
  if [foo] in [bar] {

  }
}

output {
  stdout { }
}
```

I propose that we optimize out empty `if`s in LIR. This shouldn't impact the execution of pipelines and it will simplify some LS UI tasks we're working on.

**Comment by justinkambic** [comment]
cc @ycombinator

**Comment by andrewvc** [comment]
This was resolved in #9314
